src/modules/xkeyboard.cpp

```diff
@@ -284,42 +284,45 @@ namespace modules {
 
       size_t size = vec.size();
       if (size != 2 && size != 3) {
-        m_log.warn("%s: malformed layout-icon '%s'", name(), it);
+        m_log.warn("%s: Malformed layout-icon '%s'", name(), it);
         continue;
       }
 
       const string& layout = vec[0];
       if (layout.empty()) {
-        m_log.warn("%s: layout-icon '%s' is invalid: there must always be a layout in icon declaration", name(), it);
+        m_log.warn("%s: layout-icon '%s' is invalid: there must always be a layout defined", name(), it);
         continue;
       }
 
-      if (size == 2) {
-        if (layout == drawtypes::layouticonset::VARIANT_ANY) {
-          throw module_error(
-              "Using '" + it + "' for layout-icon means declaring a default icon, use 'layout-icon-default' instead");
-        }
+      string variant;
```

Comment:
I've hesitated to make this change to avoid some string copy but being at configuration time, you're definitely right this is much nicer

Reply:
We could maye use `swap` to avoid a copy

Reply:
I think I was able to create a cleaner version of this ;)